homeassistant/components/yale_smart_alarm/config_flow.py

```diff
@@ -0,0 +1,125 @@
+"""Adds config flow for Yale Smart Alarm integration."""
+from __future__ import annotations
+
+import voluptuous as vol
+from yalesmartalarmclient.client import AuthenticationError, YaleSmartAlarmClient
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_AREA_ID, DEFAULT_AREA_ID, DEFAULT_NAME, DOMAIN, LOGGER
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_AREA_ID, default=DEFAULT_AREA_ID): cv.string,
+    }
+)
+
+
+class YaleConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Yale integration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    entry: config_entries.ConfigEntry
+
+    async def async_step_import(self, config: dict):
+        """Import a configuration from config.yaml."""
+
+        self.context.update(
+            {"title_placeholders": {CONF_NAME: f"YAML import {DOMAIN}"}}
+        )
+        return await self.async_step_user(user_input=config)
+
+    async def async_step_reauth(self, user_input=None):
+        """Handle initiation of re-authentication with Yale."""
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
```

Comment:
The entry_id hasn't been set in the context since we don't raise `ConfigEntryAuthFailed` to start the reauth step.

Reply:
I did add it to the setup_entry in init and also added it to the coordinator to be able to run reauth. Hope you can see if you think it's an ok way to deal with it.